File: Codewars/Ten-Pin_Bowling/Ten-Pin_Bowling.js
function bowlingScore(frames){
  let score = 0
  frames = frames.split(" ")
  let last = frames.slice(-1)[0]
  console.log(frames, last)
  for(let i=0 ; i < frames.length - 1; i++){
    let remaining = (frames.slice(i, frames.length)).join("")
    console.log(remaining,i)
    score += scoreRound(remaining, frames[i])
  }

  return score + scoreChars(last)
}

function scoreRound(remaining,round){
  if(round.slice(-1)[0] === "X" || round.slice(-1)[0] === "/"){
    return scoreChars(remaining.slice(0,3))
  }else{
    return parseInt(round[0]) + parseInt(round[1])
  }
}

function scoreChars(chars){
  let sum = 0
  let prev = 0
  chars = chars.split("")
  let n
  for(let i=0; i < chars.length; i++){
    if(chars[i] === "X"){
      n = 10
    }else if( chars[i] === "/"){
      n = 10 - prev
    }else{
      n = parseInt(chars[i])
    }
    prev = n
    sum += n
  }
  return sum

}

File: Codewars/Ten-Pin_Bowling/Ten-Pin_Bowling.rb
def bowling_score(frames)
  score = 0
  frames = frames.split(" ")
  last = frames[-1]
  # puts "frames #{frames}, last #{last}"
  (0..8).each do |idx|
    remaining = (frames[idx , frames.length - idx]).join()
    # puts "remaining #{remaining}"
    score += score_round(remaining, frames[idx])
  end

  return score + score_chars(last)


end

def score_round(remaining, round)
  if round[-1] == "X" or round[-1] == "/"
    return score_chars(remaining[0,3])
  else
    return round[0].to_i + round[1].to_i
  end
end

def score_chars(chars)
  sum = 0
  prev = 0
  chars.split("").each do |n|
    case n
      when "X"
        n = 10
      when "/"
        n = 10 - prev
      else
        n = n.to_i
    end
    prev = n
    sum += n
  end
  return sum
end
